### ISSUE TITLE
Multiplatform equivalent for java.util.concurrent.TimeUnit?

### ISSUE THREAD

**Issue by abissell** [korlibs/klock#7]
A lot of Kotlin devs are probably used to this API especially for conversion between different units of time. Would you be interested in a (probably very small) PR to add a multiplatform version?

**Comment by soywiz** [comment]
Yeah, I would accept PRs. But, whats the benefit over this?
```
val time = 10.seconds
val tenSecondsInMilliseconds = time.milliseconds
```

**Comment by abissell** [comment]
I think that design is great and I wouldn't want to change or replace it.

It's possible that all the changes would live in the JVM subrepo, and just be a translation layer between Klock and the Java TimeUnit stuff. For example, if I have some coroutine task with a timeout that's defined in Klock common code, it would be nice to either

**Comment by abissell** [comment]
One benefit of `TimeUnit` is that it can convert between values without any class instantiation, but that may not be enough of a gain to justify adding it to the API. Maybe all that's needed is a little translation layer within the JVM module for conversions to work with Java/Kotlin APIs which expect a `Long` and a `TimeUnit`, like [https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.experimental/with-timeout.html](withTimeout). I'll chew it over for a bit and try to get you a PR. And thanks for the great library! :)

**Comment by soywiz** [comment]
Yeah, right now it instantates objects. The same goes for operator overloads:
`1.seconds + 1000.milliseconds` instantiates 3 objects. But it is safer to do artithmetic this way, and nicer for arguments and for storing times.
Allocation will be zero once inline classes ships: https://github.com/Kotlin/KEEP/pull/103
And/Or project valhalla. Since most of these classes are immutable with a single field.

Still, if you think it makes sense, create a PR :)

**Comment by soywiz** [comment]
Klock 1.0.0 will be released later this week and will use Kotlin 1.3 inline, so I guess that this is not required anymore. I'm closing it. Feel free to reopen if required.
